client/landing/jetpack-cloud/components/sidebar/index.jsx

```diff
@@ -0,0 +1,208 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import Sidebar from 'layout/sidebar';
+import SidebarFooter from 'layout/sidebar/footer';
+import SidebarItem from 'layout/sidebar/item';
+import SidebarMenu from 'layout/sidebar/menu';
+import SidebarRegion from 'layout/sidebar/region';
+import ExpandableSidebarMenu from 'layout/sidebar/expandable';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+class JetpackCloudSidebar extends Component {
+	static propTypes = {
+		context: PropTypes.shape( {
+			path: PropTypes.string,
+		} ),
+	};
+
+	state = {
+		expandedSections: [],
+	};
+
+	componentDidMount() {
+		const section = this.getSectionFromCurrentPath();
+
+		if ( section ) {
+			this.toggleSection( section );
+		}
+	}
+
+	/**
+	 * Get current section name from path. When there's no section, we're in the dashboard.
+	 *
+	 * @returns {string} Current section
+	 */
+	getSectionFromCurrentPath() {
+		// If no section is present in the path, return 'dashboard'
+		return this.props.context.path.split( '/' )[ 2 ] ?? 'dashboard';
+	}
+
+	/**
+	 * Get current subsection name from path.
+	 *
+	 * @returns {string|null} Current subsection or null
+	 */
+	getSubsectionFromCurrentPath() {
+		return this.props.context.path.split( '/' )[ 3 ] ?? null;
+	}
+
+	/**
+	 * Toggle an expandable section.
+	 *
+	 * @param   {string} section Section to be toggled
+	 * @returns {void}
+	 */
+	toggleSection( section ) {
+		if ( this.isExpanded( section ) ) {
+			this.setState( {
```

Comment:
Since we're setting the state in both case, should we keep the `setState` just once, and move the different logic outside? Like this (untested):

```
const expandedSections = this.isExpanded( section )
	? this.state.expandedSections.filter( item => item !== section )
	: [ ...this.state.expandedSections, section ];
this.setState( { expandedSections } );
```

Reply:
Sounds good 👍

Reply:
Addressed in 603c49e.